llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -8063,7 +8079,8 @@ AArch64TargetLowering::LowerCall(CallLoweringInfo &CLI,
   // return.
   SDValue Result = LowerCallResult(Chain, InGlue, CallConv, IsVarArg, RVLocs,
                                    DL, DAG, InVals, IsThisReturn,
-                                   IsThisReturn ? OutVals[0] : SDValue());
+                                   IsThisReturn ? OutVals[0] : SDValue(),
+                                   RequiresSMChange.has_value());
```

Comment:
RequiresSMChange being an `optional<bool>` is unclear to me what it means, and why having any value at all is what you want for the value of LowerCallResult()'s `bool RequiresSMChange` parameter.

Reply:
That interface is indeed awfully confusing! I've refactored it here: #78703